### PR TITLE
Fix quality settings UI and shadow mask UI

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
Fix for : https://fogbugz.unity3d.com/f/cases/1232042/
+ 
Fix for : https://fogbugz.unity3d.com/f/cases/1232011/

It rely on C++ PR: https://ono.unity3d.com/unity/unity/pull-request/104532/_/graphics/quality-settings-ui

This PR hide shadow mask property in quality settings as well as realtime reflection probe property.
It also enable/disable UX correctly when Shadow Mask is support is disabled in HDRP asset.

Test:
Checked that behavior is correct when switching between different pipeline and with several pipeline in quality settings and switching to them. Yamato should handle the shadow mask still working part.

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 1]
```diff
@@ -102,6 +102,12 @@ internal static Volume GetOrCreateDefaultVolume()
         readonly PostProcessSystem m_PostProcessSystem;
         readonly XRSystem m_XRSystem;
 
+        // Keep track of previous Graphic and QualitySettings value to reset when switching to another pipeline
+        bool m_previousLightsUseLinearIntensity;
+        bool m_previoulightsUseColorTemperature;
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs:107]: Typo: m_PreviousLightsUseColorTemperature

**Comment by sebastienlagarde** [comment]
> Small naming issues

Fixed

**Comment by iM0ve** [comment]
* The Realtime probe and Shadowmask Mode are hidden in Quality settings when HDRP asset is assigned. Again visible when HDRP asset is unassigned 

Issues:
* Error is being spammed whenever Shadowmask is disabled in HDRP asset. Reproduction:
  1) [Optional] Download project from case 1232011
  2) [Optional] Run HDRP wizard
  3) Toggle the Shadowmask checkbox in the HDRP asset
  4) Notice the error

image

**Comment by sebastienlagarde** [comment]
@iM0ve could you check if the error happen in HDRP/staging already? and also if this error happen with any other option (I don't think it is related to shadowmask, but a more general issue). thanks

**Comment by iM0ve** [comment]
Yes appears to be non PR specific. Also on staging. I updated the comment. Also it happens with other checboxes like Ambient Occlusion, Volumetrics, etc

**Pull request merged**